internal/job/currency-job.go

```diff
@@ -0,0 +1,20 @@
+package job
+
+import (
+	"genesis-currency-api/internal/service"
+	"github.com/robfig/cron/v3"
+	"log"
+	"time"
+)
+
+// UpdateCurrencyJob is a cron function to update currency service cache.
+// It is executed every hour.
+func UpdateCurrencyJob(currencyService *service.CurrencyService) {
+	scheduler := cron.New(cron.WithLocation(time.UTC))
```

Comment:
You create cron instances in UpdateCurrencyJob and SendEmailsJob. In result, you have two the same objects when you can only have one (and, as i understood, cron object can register multiple functions). In my opinion, UpdateCurrencyJob and SendEmailsJob should be those anonymous functions, that must be registered in StartAllJobs function.

 Moreover, the Dependency Injection principle is violated here. Ideally you should expect cron object as a argument of StartAllJobs.